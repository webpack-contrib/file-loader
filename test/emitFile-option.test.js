import {
  compile,
  execute,
  getCompiler,
  normalizeErrors,
  readAsset,
} from './helpers';

describe('"emitFile" option', () => {
  it('should work without value', async () => {
    const compiler = getCompiler('simple.js');
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(Object.keys(stats.compilation.assets)).toMatchSnapshot('assets');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });

  it('should work with "Boolean" value equal "true"', async () => {
    const compiler = getCompiler('simple.js', {
      emitFile: true,
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(Object.keys(stats.compilation.assets)).toMatchSnapshot('assets');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });

  it('should work with "Boolean" value equal "false"', async () => {
    const compiler = getCompiler('simple.js', {
      emitFile: false,
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(Object.keys(stats.compilation.assets)).toMatchSnapshot('assets');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });
});
