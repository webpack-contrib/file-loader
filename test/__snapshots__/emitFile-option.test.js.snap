// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"emitFile" option should work with "Boolean" value equal "false": assets 1`] = `
Array [
  "main.bundle.js",
]
`;

exports[`"emitFile" option should work with "Boolean" value equal "false": errors 1`] = `Array []`;

exports[`"emitFile" option should work with "Boolean" value equal "false": result 1`] = `"9c87cbf3ba33126ffd25ae7f2f6bbafb.png"`;

exports[`"emitFile" option should work with "Boolean" value equal "false": warnings 1`] = `Array []`;

exports[`"emitFile" option should work with "Boolean" value equal "true": assets 1`] = `
Array [
  "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  "main.bundle.js",
]
`;

exports[`"emitFile" option should work with "Boolean" value equal "true": errors 1`] = `Array []`;

exports[`"emitFile" option should work with "Boolean" value equal "true": result 1`] = `"9c87cbf3ba33126ffd25ae7f2f6bbafb.png"`;

exports[`"emitFile" option should work with "Boolean" value equal "true": warnings 1`] = `Array []`;

exports[`"emitFile" option should work without value: assets 1`] = `
Array [
  "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  "main.bundle.js",
]
`;

exports[`"emitFile" option should work without value: errors 1`] = `Array []`;

exports[`"emitFile" option should work without value: result 1`] = `"9c87cbf3ba33126ffd25ae7f2f6bbafb.png"`;

exports[`"emitFile" option should work without value: warnings 1`] = `Array []`;
